### PR TITLE
Typescript: use outFile and files options with tsconfig.json

### PR DESCRIPTION
**Pull request by AlexVangelov**

### CODE REVIEW

**Comment by ggoodman** [comment]
Very interesting. Thanks for this great work. I'll take a look ASAP.

File: adapters/transformers/typescript.js [hunk 1]
```diff
@@ -15,6 +15,15 @@ var defaultCompileOptions = _.extend(Typescript.getDefaultCompilerOptions(), {
 module.exports = {
   matches: /\.ts$/,
   provides: ".js",
+  providesIndirect: function(pathname, files) {
```
Comment (ggoodman) [inline adapters/transformers/typescript.js]: Very interesting approach. Likely applicable for your experiments w/ `webpack`.

Can you think of other uses that might need additional metadata to make a decision? I wonder if `function(requestPath, files)` is the right signature.

**Comment by ggoodman** [comment]
LMK what your thoughts are on the `providesIndirect` api question. Otherwise LGTM and a huge thanks!

**Comment by AlexVangelov** [comment]
Looks like files content is enough to get the decision for provideIndirect.
But we can provide some feedback to the user in case of error through `preview` object.
See https://github.com/ggoodman/plunker-run-plugin/commit/eb5a6d4829971bfe69e44737a70b1661b41d9d70 modification.
Do you have a suggestion?

**Comment by ggoodman** [comment]
@AlexVangelov The logic in this server is unnecessarily complicated so I did a partial rewrite with a much-simplified transformer logic. Here is what the updated `less` transformer looks like: https://github.com/ggoodman/plunker-run/blob/master/facets/renderer/lib/lessRenderer.js

Essentially, each renderer is a module exporting a `{ name, getRenderer }` tuple where `getRenderer` is a function with the following signature:

``` js
function (preview, pathname) { }
```

**`preview`** is an object representing the current preview and its entries as per: https://github.com/ggoodman/plunker-run/blob/master/facets/previews/lib/preview.js

**`pathname`** is the path and filename that is being requested.

The `getRenderer` function must _synchronously_ check if it can provide `pathname` given the contents of `preview`. If it can, it must return a `render()` function that will actually do the transformation. The `render` function can return a `Promise` that resolves to an object with `{ headers, encoding, payload }`.

What do you think?

**Comment by AlexVangelov** [comment]
Sounds good. It's clean.
I will try the new approach with the Webpack experiments.

PS: https://plnkr.co is down right now... redirected to https://my.modulus.io/project-not-found?url=plnkr.co / NO APPLICATION WAS FOUND FOR plnkr.co

**Pull request closed without merging**